node/test/Test/Pos/Network/Windows/DnsDomainsSpec.hs

```diff
@@ -14,23 +15,35 @@ module Test.Pos.Network.Windows.DnsDomainsSpec
 import           Universum
 
 import           Network.URI                    (isIPv4address)
-import           Test.Hspec                     (Expectation, Spec, describe,
-                                                 shouldSatisfy)
+import           Pos.Network.Types              (initDnsOnUse)
+import           Test.Hspec                     (Expectation, Spec, describe, it,
+                                                 shouldBe, shouldSatisfy)
 import           Test.Hspec.QuickCheck          (prop)
 
 #if !defined(POSIX)
 import qualified Pos.Network.Windows.DnsDomains as Win
 #endif
 
+
+testLookupFor :: ByteString -> Expectation
+testLookupFor dnsDomain = do
+    initDnsOnUse $ \resolve -> do
+      res <- resolve dnsDomain
+      isRight res `shouldBe` True
+
 spec :: Spec
 #if defined(POSIX)
 spec = return ()
 #else
-spec =
+spec = do
     describe "Default DNS server on Windows" $ do
         describe "getWindowsDefDnsServer" $ do
             prop "Successfully retrieves this Windows machine's DNS server"
                  prop_GetWindowsDefDnsServer
+    describe "Multi-value lookups" $ do
+        it "www.google.it resolves correctly" $ testLookupFor "www.google.it"
+        it "cardano-node-0.aws.iohkdev.io resolves correctly" $ testLookupFor "cardano-node-0.aws.iohkdev.io"
+        it "cardano-node-1.aws.iohkdev.io resolves correctly" $ testLookupFor "cardano-node-1.aws.iohkdev.io"
```

Comment:
This may not be a robust test since it depends on our devops DNS setup.

Also, this does not really test multi-valued lookup.